backend/apps/matching/configuration/properties.py

```diff
@@ -0,0 +1,244 @@
+from itertools import chain
+import string
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+import numpy as np
+
+
+class Property:
+
+    properties = None
+
+    def __init__(self, label=None, name=None, help_text=None, info_text=None, private=False):
+        """
+        Create a new property of a participant.
+
+        label: Label of the property that is shown in the UI
+        help_text: Help text that should be displayed on hover in the UI
+        info_text: Information that is visible next to the property in the UI
+        private: Is true if only site administrators should be able to view this property
+        """
+        if label is None:
+            raise ValueError("You need to set the value for label.")
+        self.label = _(label)  # the text that should appear when this property is referenced
+        self.name = name if name is not None else label.lower().replace(" ", "_")
+        if "--" in self.name:
+            raise ValueError(
+                f"The field name {self.name} is not allowed, since it contains '--'."
+                " You can explicitly chose a suitable column name by setting the 'name' kwarg."
+            )
+        if len(self.name) > 12:
+            raise ValueError(
+                f"The field name {self.name} is too long. You can use 12 characters at max."
+                f" If you did not set the 'name' before, please set in in the constructor. "
+            )
+        self.help_text = help_text
+        self.info_text = info_text
+        self.private = private
+
+    def get_model_field_names(self, prefix=None):
+        raise NotImplementedError
+
+    def get_model_fields(self):
+        raise NotImplementedError
+
+    def generate_random_assignment(self, rs=None):
```

Comment:
It's hard to review without a documentation string like the nice one for `__init__`. I will stop here for the moment.

Reply:
I just pushed an updated version with more detailed comments. Could you try to review again.

Maybe start at the 'A.py' where the config is defined. I think this will make everything more clear.